linux/debian/man/km-package-install.1
.\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.48.4.
.TH KM-PACKAGE-INSTALL "1" "July 2021" "km-package-install version 15.0.81" "User Commands"
.SH NAME
km-package-install \- Install a Keyman keyboard package
.SH SYNOPSIS
.B  km-package-install
 [\fI\,-h\/\fR] [\fI\,-s\/\fR] [\fI\,-f KMPFILE\/\fR] [\fI\,-p PACKAGE\/\fR] [\fI\,--version\/\fR] [\fI\,-v\/\fR]
.SH DESCRIPTION
.IP
[\-vv]
.PP
Install a Keyman keyboard package, either a local .kmp file or specify a
keyboard id to download and install
.SS "optional arguments:"
.TP
\fB\-h\fR, \fB\-\-help\fR
show this help message and exit
.TP
\fB\-s\fR, \fB\-\-shared\fR
Install to shared area \fI\,/usr/local\/\fP
.TP
\fB\-f\fR KMPFILE, \fB\-\-file\fR KMPFILE
Keyman kmp file
.TP
\fB\-p\fR PACKAGE, \fB\-\-package\fR PACKAGE
Keyman package id
.TP
\fB\-\-version\fR
show program's version number and exit
.TP
\fB\-v\fR, \fB\-\-verbose\fR
verbose logging
.TP
\fB\-vv\fR, \fB\-\-veryverbose\fR
very verbose logging
.SH "TAB COMPLETION"

You may use tab completion to find keyboard packages to install.
The list comes from the network so there may be a small delay before
tab completion will work the first time you use it.
The list of packages is cached for one week.

The list is the full list including keyboards that do not work on
Linux, so they won't necessarily install.
.SH "SEE ALSO"
km-config(1)
km-package-list-installed(1)
km-package-uninstall(1)


HISTORY.md
# Keyman Version History

## 14.0.280 stable 2021-08-05

* chore(ios,windows): Update crowdin strings for Azerbaijani 🍒  (#5487)

## 14.0.279 stable 2021-07-21

* fix(ios): keyboard swapping 🍒 (#5476)

## 14.0.278 stable 2021-07-20

* change(common/core/web): fat-finger-performance unit-test threshold 🍒 (#5417)
* fix(linux): Improve bitmap conversion 🍒 (#5410)
* fix(linux): Save QR code in temporary directory 🍒 (#5411)
* fix(linux): Fix restarting ibus when running with `sudo` 🍒 (#5413)
* fix(linux): Don't crash with corrupt keyboard 🍒 (#5421)
* fix(web): fixes osk resize-popping effect (#5428)
* fix(android/engine): register lexical model in switchToNextKeyboard 🍒  (#5447)
* feat(web): spacebar text controls 🍒 🌌 (#5406)
* feat(android): Spacebar text controls 🍒 🌌 (#5407)
* feat(ios): add spacebar text controls 🍒 🌌 (#5408)
* chore(android): improve error reporting for kmw 🍒 (#5469)

## 14.0.277 stable 2021-06-29

* feat(android): silent install for referred kbd 🍒 (#5266)
* chore(common): bash version for run-required-test-builds.sh 🍒 (#5268)
* chore(common): Add YouTube links to Keyman 14 features 🍒  (#5285)
* fix(common/models): predictive-text engine use of NFD input 🍒 (#5286)
* fix(developer): kmconvert help match reality 🍒 (#5299)
* chore(linux): Update changelogs to match Debian (#5303)
* fix(mac): .kmx max file version is now 0x0E00 (14.0) 🍒 (#5331)
* fix(mac): kmp.inf is windows-1252 🍒 (#5330)
* fix(windows): improve keymanx64 start stability 🍒 (#5222)
* chore(linux): Build also on riscv64 to facilitate migration on Ubuntu (#5322)
* fix(android): fixes application of nextlayer for subkeys with customized layer setting 🍒 (#5351)
* feat(common/core/web): engine correction-prep optimizations 🍒 (#5352)
* fix(linux): Add libglib2.0-bin to keyman Depends (#5359)
* fix(windows): osk scaling mismatch on horz axis 🍒 (#5342)
* fix(linux): Don't add duplicate entries when reinstalling keyboard 🍒 (#5369)
* chore: keyboard_info.source.json 1.0.6 🍒 (#5380)
* feat(android): hide textarea for perf 🍒 (#5377)
* change(ios): setText now uses JSON serialization 🍒 (#5333)

## 14.0.276 stable 2021-06-11

* fix(linux): Exclude s390x from package builds for ibus-keyman 🍒 (#5220)
* fix(web): Always pass kill_browserstack 🍒  (#5235)
* feat(android): Automatically install keyboard through Play Store 🍒 (#5231)
* fix(oem/fv/ios): app encryption flag for app store uploads (14.0) (#5243)
* fix(common/core/web): optimizes transform construction 🍒 (#5255)
* fix(android): rotation is not updating keyboard 🍒 (#5262)
* feat(android): silent install for referred kbd 🍒 (#5266)
* fix(windows): disable new hotkey modifier check (#5270)

## 14.0.275 stable 2021-06-04

* fix(linux): Swap order of dependencies for Debian package 🍒 (#5070)
* fix(android/engine): Check lexical-model file exists before using 🍒  (#5080)
* fix(developer): kmconvert commandline and deploy 🍒 (#5083)
* fix(web): position popups correctly in landscape mode on Android and during Chrome emulation 🍒 (#5090)
* fix(common/core/desktop): Fix warnings when compiling for armhf 🍒 (#5109)
* fix(common/core/desktop): Don't segfault on invalid .kmx file 🍒 (#5110)
* chore(common): Check in crowdin strings for Amharic 🍒  (#5117)
* chore(linux): Swap order of dependency 🍒 (#5114)
* fix(developer): TframeTextEditor.SetText was not synchronous 🍒 (#5097)
* chore(ios): update fv cert ref 🍒 (#5124)
* chore(ios): works around wrong-workspace Carthage 0.38 lookup issue (#5135)
* fix(web): multi touch handling 🍒 (#5142)
* chore(ios): artifact prep script now in-repo (#5141)
* chore(web): Add CI script to kill BrowserStack tunnel 🍒  (#5152)
* fix(ios): artifact upload preparation 🍒 (#5161)
* chore(android): Add -clean flag to build script 🍒  (#5177)
* fix(windows): avoid cached hotkey state 🍒 (#5189)
* fix(android/samples): Set gradlew executable for Tests 🍒  (#5199)
* fix(common/core/desktop): Fix failing tests on armhf 🍒 (#5204)
* feat(web): Add script to deploy KeymanWeb release to s.keyman.com 🍒  (#5179)
* chore(ios): renames problematic app selectors 🍒 (#5174)
* fix(developer): support spaces in regression tests 🍒 (#5218)

## 14.0.274 stable 2021-05-11

* fix(android/engine): Fix toHex() for null string 🍒  (#4997)
* chore: support for xcode 12 🍒 (#4996)
* chore(linux): Add support for Ubuntu 21.04 (hirsute) 🍒 (#5035)
* chore(ios): prep for CI transition to Xcode 12, build script tweak 🍒 (#5048)
* chore(windows): add telemetry to trace crash on exit 🍒 (#5008)
* fix(windows): avoid disabling Keyman when speech recognition starts 🍒 (#5038)
* chore(android/app): Always use offline help 🍒  (#5051)
* chore(common): Check in crowdin files for Azerbaijani 🍒  (#5039)
* feat(android/app): Add telemetry for launching WebBrowserActivity 🍒  (#5050)
* fix(linux): Enable caps lock support 🍒 (#5058)

## 14.0.273 stable 2021-04-26

* chore(linux): Fix triggering of Jenkins builds for stable branch (#4968)
* fix(linux): Don't crash if kmp file vanishes (fixes #4907) 🍒 (#4972)
* fix(linux): Fix crash with incomplete metadata (#4908) 🍒 (#4979)
* chore(android/engine): Don't use localized string for Sentry errors 🍒  (#4980)
* chore: disable findTouchAliasElement logging 🍒 (#4982)
* fix(web): Make banner initialization more robust 🍒  (#4983)
* chore: Add cherry-pick label for cherry-pick PRs 🍒 (#4984)

## 14.0.272 stable 2021-04-22

* chore(linux): Debug triggering Jenkins build (#4852)
* chore(android/engine): Rename "Fula" to "Pulaar" 🍒  (#4860)
* chore(android/engine): Rename "Pulaar" to "Pulaar-Fulfulde" 🍒  (#4891)
* fix(developer): Reduce non-canonical BCP 47 tag warning in PackageInfo to Info 🍒  (#4892)
* chore(common): Update crowdin for French 🍒  (#4903)
* chore(android,windows): Check in crowdin for Indonesian 🍒  (#4904)
* fix(linux): don't crash on legacy non-Unicode files (#4878)
* fix(android/engine): Don't load woff fonts on Android N 🍒  (#4924)
* fix(linux): Don't crash on network problem (fixes #4911) 🍒 (#4931)

linux/help/reference/km-package-install.php
<?php
require_once('includes/template.php');
require_once('includes/__docheader.php');

$baselanguage = "language-bash";

head([
  'title' => "Command: km-package-install"
]);

?>

<!-- TODO: Auto-generate from https://manpages.debian.org/unstable/keyman/km-package-install.1.en.html -->
<div class="body_text">
  <h1 class="title" id="kmPackageInstall">km-package-install</h1>

  <h2 id="Summary" name="Summary">Summary</h2>

  <p>Install a Keyman keyboard package</p>

  <?php
  syntax("km-package-install [-h] [-s] [-f <kmpfile>] [-p <packageID>] [--version] [-v] [-vv]");
  params("-h, --help", "optional", "show this help message and exit",
    "-s, --shared", "optional", "install to shared area <code>/usr/local</code>",
    "-f &lt;kmpfile&gt;, --file &lt;kmpfile&gt;", "optional", "Keyman kmp file",
    "-p &lt;packageID&gt;, --package &lt;packageID&gt;", "optional", "Keyman package id",
    "--version", "optional", "show program's version number and exit",
    "-v, --verbose ", "optional", "verbose logging",
    "-vv, --veryverbose", "optional", "very verbose logging");
  ?>

  <h2 id="Description" name="Description">Description</h2>
  <p>km-package-install installs a Keyman keyboard. Provide either a local .kmp file or specify a package id
    to download and install.
  </p>
  <h3>Tab Completion</h3>
  <p>
    You may use tab completion to find keyboard packages to install. The list comes from the network,
    so there may be a small delay before tab completion will work the first time you use it. The list of packages
    is cached for one week.
  </p>
  <p>
    The list is the full list including keyboards that do not work on Linux, so they won't ncecessarily install.
  </p>

  <h2 id="See_also" name="See_also">See also</h2>
  <ul>
    <li><a href="km-config.php" title="km-config"><code>km-config</code></a></li>
    <li><a href="km-package-list-installed.php" title="km-package-list-installed"><code>km-package-list-installed</code></a></li>
    <li><a href="km-package-uninstall.php" title="km-package-uninstall"><code>km-package-uninstall</code></a></li>
  </ul>

</div>
